### ISSUE TITLE
find_work_for_edition misses variant author spellings, causing redundant works

### ISSUE THREAD

**Issue by LeadSongDog** [internetarchive/openlibrary#667]
This does not check for variant spellings of authors when multiple author records exist for a given author: 

https://github.com/internetarchive/openlibrary/blob/7afe9d853bdb282a70c11ff80b2a3bba874e61ac/openlibrary/catalog/works/find_work_for_edition.py#L14

The result is that often there is no matching work returned when there should be.

Similarly, variations in title spelling (accents, capitalization, or omission of stopwords) seem to cause the same effect.

**Comment by tfmorris** [comment]
I think using the author key (ie the current method) is the only realistic way to do this. Doing simple string matching will result in too many false matches. The things that needs to be done to make this work, though, is to get all the duplicate author records merged. 

The title normalization is done in mk_norm here: https://github.com/internetarchive/openlibrary/blob/7afe9d853bdb282a70c11ff80b2a3bba874e61ac/openlibrary/catalog/utils/__init__.py#L290 

It strips leading "A" & "The" and bracketed text as well collapsing whitespace, but it could be better about handling:

- non-English leading articles
- case, diacritic, and normalization form variations
- subtitles

For it to correctly handle editions which are translations, it will need significant enhancement.

**Comment by mekarpeles** [comment]
Is the real problem that we have too many duplicate authors which need to get merged?

**Comment by LeadSongDog** [comment]
@mekarpeles I think that is a real problem but this one blocks that one. We can’t merge author records until we identify them as having the same author referent. Search results ought to find near misses and highlight the differences.

**Comment by tfmorris** [comment]
No. Nothing this low level should be worrying about author name spellings.

**And**, bug reports should describe behavior that needs to be changed or improved, not pointing to tidbits of Python code (unless in the form of a pull request which fixes the problem and doesn't introduce others).

It sounds like the fundamental problem is that the process of finding a matching work for a given edition gives too many false negatives. Is that correct? How do we want to bias the false negatives vs false positives equation?

**Comment by LeadSongDog** [comment]
@tfmorris
Be that as it may, when both the title spellings and author spellings are very similar between an existing work record and a newly added edition, the default behaviour should _not_ be to autocreate a new work record, nor a new author record. If we can't get it automated correctly, then check with a human!

**Comment by LeadSongDog** [comment]
As @tfmorris points out, handling of case, diacritics, and normalization forms still needs improvement. 

Æsop, Aesop, Äsop, Esop, and Αἴσωπος etc. should all match the same author record without resort to locally creating a manually curated a.k.a. list. That is what the curated ISNI is for. Our various multinational source catalogues could have used any or all of the 65 spellings found there, and we have no agreed way to prioritize one over the others. Without curation, VIAF is a poor substitute, as it frequently conflates homologous identities.

Before creating any new author record, we should run the name through a search equivalent to: https://isni.oclc.org/xslt/DB=1.2/SET=2/TTL=1/CMD?ACT=SRCH&IKT=3003&SRT=LST_nd&TRM=Aesop
This would identify the most common variations, any of which might correspond to an existing OL author record. These could (or should) be added to the a.k.a. list.

Finding the correct extant author record will preclude a large portion of redundant new work creations.

**Comment by jimman2003** [comment]
ISNI can be also found throgh wikidata sometimes
